### ISSUE TITLE
Bug - Water does not properly freeze/thaw as seasons change

### ISSUE THREAD

**Issue by frivolour** [RustyMods/Seasonality#17]
TLDR: At the beginning of winter, ice will not form near the player until you relog. At the end of winter, ice will not thaw until you restart your game.

Steps to reproduce freezing behavior:
1. Set Winter Freezes Water setting to On in config.
2. Load into a world, find some water, stand next to it.
3. Wait for winter to come (or set season to winter).
    - Once winter starts, the water near the player will not be frozen. However, water further out will be. I didn't measure the distance exactly but it seems to roughly correspond to the 64x64 chunk the player is standing in. 
    - Relogging at this point will fix the issue and the water will become frozen as expected

Steps to reproduce thawing behavior:
1. Set Winter Freezes Water setting to On in config.
2. Load into a world, set season to winter.
3. Relog, to ensure all water is frozen.
5.  Find some water, and stand next to it.
6. Set season to spring.
    - The water next to the player will remain frozen.
    - As you move from where you were standing when the seasons changed, water will freeze as it is loaded in (so unlike with freezing, all water is in the incorrect state rather than just the stuff near you).
    - Relogging at this point will not fix the issue, the water remains frozen. However, restarting the game will fix it and thaw the water.

Seasonality version: 3.1.6
BepInExPack_Valheim version: 5.4.2202
Valheim version: 0.217.38
Launcher: r2modman v3.1.46
